docs/advanced/customization-guide.md

```diff
@@ -16,8 +15,1007 @@ published: false
+following stdout output, or alternatively, a plaintext file
+`/etc/kubernetes/node-feature-discovery/features.d/my-features` having the
+following contents:
+
+```plaintext
+my-feature.1
+my-feature.2=myvalue
+my.namespace/my-feature.3=456
+```
+
+This will translate into the following node labels:
+
+```yaml
+feature.node.kubernetes.io/my-feature.1: "true"
+feature.node.kubernetes.io/my-feature.2: "myvalue"
+my.namespace/my-feature.3: "456"
+```
+
+Note that in the example above `-extra-label-ns=my.namespace` must be specified
+on the nfd-master command line.
+
+### Hooks
+
+The *local* source executes hooks found in
+`/etc/kubernetes/node-feature-discovery/source.d/`. The hook files must be
+executable and they are supposed to print all discovered features in `stdout`.
+With ELF binaries static linking is recommended as the selection of system
+libraries available in the NFD release image is very limited. Other runtimes
+currently supported by the NFD image are bash and perl.
+
+`stderr` output of hooks is propagated to NFD log so it can be used for
+debugging and logging.
+
+NFD tries to execute any regular files found from the hooks directory.
+Any additional data files the hook might need (e.g. a configuration file)
+should be placed in a separate directory in order to avoid NFD unnecessarily
+trying to execute them. A subdirectory under the hooks directory can be used,
+for example `/etc/kubernetes/node-feature-discovery/source.d/conf/`.
+
+**NOTE:** NFD will blindly run any executables placed/mounted in the hooks
+directory. It is the user's responsibility to review the hooks for e.g.
+possible security implications.
+
+**NOTE:** The [minimal](deployment-and-usage#minimal) image variant only
+supports running statically linked binaries.
+
+### Feature files
+
+The *local* source reads files found in
+`/etc/kubernetes/node-feature-discovery/features.d/`.
+
+### Input format
+
+The hook stdout and feature files are expected to contain features in simple
+key-value pairs, separated by newlines:
+
+```plaintext
+<name>[=<value>]
+```
+
+The label value defaults to `true`, if not specified.
+
+Label namespace may be specified with `<namespace>/<name>[=<value>]`. The
+namespace must be explicitly allowed with the `-extra-label-ns` command line
+flag of nfd-master if using something else than
+`[<sub-ns>.]feature.node.kubernetes.io` or
+`[<sub-ns>.]profile.node.kubernetes.io`.
+
+### Mounts
+
+The standard NFD deployments contain `hostPath` mounts for
+`/etc/kubernetes/node-feature-discovery/source.d/` and
+`/etc/kubernetes/node-feature-discovery/features.d/`, making these directories
+from the host available inside the nfd-worker container.
+
+#### Injecting labels from other pods
+
+One use case for the hooks and/or feature files is detecting features in other
+Pods outside NFD, e.g. in Kubernetes device plugins. By using the same
+`hostPath` mounts for `/etc/kubernetes/node-feature-discovery/source.d/` and
```

Comment:
it's not NFD specific as such but on SELinux enabled platforms it's more than just `hostPath`. Maybe add a note here that it needs to be taken into account too or, preferably, the suggestion what needs to be done to overcome that limitation.

Reply:
TBH, I haven't tried this in an SELinux-enabled environment. Any suggestion what to add? *Note that if SELinux is enabled you need to correctly set `seLinuxOptions` in the security context of all pods mounting the same host path.*(?)

Reply:
I was thinking @ArangoGutierrez might be able to help with that.

Reply:
OpenShift has SELinux enabled, or you can enable it on MiniKube for testing.
Whit that said, yes, I usually get around that by running the 2 containers in the same pod, that way they share namespace and you can easily share mounts. (This for documentation comes handy we don't want people to think they can run pod's on other nodes or something, it has to be same node, and preferably same pod)

Reply:
This doc talks about pods outside nfd. My comment was should we add note that they need to run with some specific selinux label to be able to write to nfd dirs?